packages/api/src/test-server.ts

```diff
@@ -94,11 +94,19 @@ export default Promise.resolve().then(async () => {
     db: server.db,
     queue: server.queue,
   };
-});
+}
 
 afterAll(async () => {
   if (server) {
+    await server.queue.close();
     await server.close();
+    server = null;
   }
   fs.removeSync(dbPath);
 });
+
+type UnboxPromise<T> = T extends Promise<infer U> ? U : T;
+
+export type TestServer = UnboxPromise<ReturnType<typeof setupServer>> &
```

Comment:
🤯

Reply:
Right? I haven't done much of that "type programming" before but it's really powerful. I think even the kebab=>camel case thing may be possible with newer typescript features ([e.g.](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/39255#issuecomment-763091386)), but I'll have to take another look on that later. Certainly let me know if I'm exaggerating at any point 😆